src/dbnode/storage/database.go

```diff
@@ -292,22 +292,15 @@ func (d *db) UpdateOwnedNamespaces(newNamespaces namespace.Map) error {
 		}
 	}
 
-	d.Lock()
-	defer d.Unlock()
-
+	d.RLock()
 	removes, adds, updates := d.namespaceDeltaWithLock(newNamespaces)
 	if err := d.logNamespaceUpdate(removes, adds, updates); err != nil {
 		enrichedErr := fmt.Errorf("unable to log namespace updates: %v", err)
 		d.log.Error(enrichedErr.Error())
+		d.RUnlock()
```

Comment:
Could we move `RUnlock` few lines up (before `logNamespaceUpdate`)?
We are using sync logging so it might be better to do outside of this coarse lock if possible.
Also then just one `RUnlock` would be enough.

Reply:
Sure, this should be safe to change.